hr_holidays_public/models/resource_calendar.py

```diff
@@ -52,12 +52,16 @@ def _public_holidays_leave_intervals(self, start_dt, end_dt, employee_id,
                             line.date,
                             time.max
                         ).replace(tzinfo=tz),
-                        line
+                        leave_obj.new({
+                            'date_from': datetime.combine(
+                                line.date, time(0, 0, 0)),
```

Comment:
This is not taking tzinfo. You can extract previous computations into variables for assigning them several times.

Reply:
As you pointed. An passing an empty object does the trick as well